File: http_request_form/python2/http_request_form.py
#!/usr/bin/env python2

import requests

url = 'http://pgp.mit.edu/pks/lookup'
vals = {
    'search': 'jmmahler@gmail.com',
    'op': 'index',
}

res = requests.get(url, params=vals)

if 'Jeremiah Mahler' in res.text:
    print("OK")
else:
    print(res.url)
    print(res.text)
    print("ERROR Unexpected response")

File: http_request_form/perl/http_request_form.pl
#!/usr/bin/perl

use LWP::Simple;
use URI;

my $url = 'http://pgp.mit.edu/pks/lookup';
my %form_vals = (
	search => 'jmmahler@gmail.com',
	op => 'index',
);

my $uri = URI->new($url);
$uri->query_form(%form_vals);

my $res = get $uri;

if ($res =~ /Jeremiah Mahler/) {
	print "OK\n";
} else {
	print "ERROR Unexpected response\n";
}
